### ISSUE TITLE
UF 4.2.0 changes to migration naming convention should only include PHP files

### ISSUE THREAD

**Issue by elliottpost** [userfrosting/UserFrosting#965]
Since UF `4.2.0`, the migration naming convention has been relaxed. This change means that any file inside the migration directories will be picked up by the migrator. I would suggest that **since the migrator is expected to only look for PHP classes, that it limits the search pattern to include only PHP files**. If the migrator encounters a SQL file or CSV file (that our project was using for seeding and raw DB queries for updating tables with enum, for instance [which Eloquent doesn't support]), it gives an error like:

```bash
 [ERROR] Unable to find the migration class
         '\UserFrosting\Sprinkle\CardManager\Database\Migrations\v20171115\brand_ids'.
```

We are unable to run any migrations since updating to `4.2.0` without moving any migration helper files out of the migration directories and updating the old migrations to find a new path. It's not preventing us from working since a work-around is known, but keeping the data files for migrations in the same directory as the migration itself would be helpful.

You can recreate this by adding any non-PHP file to any migration subdirectory.

**Comment by elliottpost** [comment]
Quick update to note that I recognize seeds no longer should live under migrations, but I would still argue that if the migrator is expecting PHP classes it should only look at class files, which allows more flexibility with the migration tool itself to allow helper assets.

**Comment by lcharette** [comment]
That's indeed a good idea. We could have any king of garbage files in there (`.gitkeep`, .DS_STORE) that could break everything.

**Comment by elliottpost** [comment]
It looks like the best place to approach this would be with the `UserFrosting\UniformResourceLocator\ResourceLocatorInterface` and `UserFrosting\UniformResourceLocator\ResourceLocator` to accept some sort of `glob()` like filter that would add it to the `$list[]` array in this snippet https://github.com/userfrosting/UniformResourceLocator/blob/master/src/ResourceLocator.php#L420-L427

I see you're the author of that package, but I recognize you're quite busy with the rest of UF as well. If I get a chance over the next week or so I'll try submitting a PR for this myself. Might be a good chance to get a better understanding of the file system tool used in UF.

**Comment by elliottpost** [comment]
PR added for URL repository at https://github.com/userfrosting/UniformResourceLocator/pull/9. 

Another PR will need to be submitted for UF core's migration system to add blacklisted common extensions (I assumed a blacklist would be better than a whitelist), and a config could easily be added to account for blacklisting by sprinkle, like:
`migration.blacklistedExtensions = ['csv', 'json', ...]`

**Comment by lcharette** [comment]
In this case, I guess URI should have both options (white and blacklist). If we want only to load php file, we don't want to blacklist all of the other don't we? We could have a `.png` in there that might break things...

**Comment by elliottpost** [comment]
I see your point. I have to leave town for a couple weeks but if this is still open when I come back I'll follow-up on the original PR (above) with your recommendations/thoughts and try again.

**Comment by xrobau** [comment]
https://github.com/userfrosting/UserFrosting/pull/998 resolves this in a simple way,

**Comment by lcharette** [comment]
While I like the whitelist/blacklist option better, for a short term solution #998 can indeed solve this. I'll leave this one open, as it's more of a long term solution.

**Comment by lcharette** [comment]
See comment here : https://github.com/userfrosting/UniformResourceLocator/pull/9

Initial issue should be solved in next version by #998.
